### PR TITLE
Quail board addition

### PR DESCRIPTION
**Pull request by MikroBusNet**
Addition of Quail board, using a new branch.

Signed-off-by: Christophe Gerbier <christophe@mikrobusnet.org>

### CODE REVIEW

File: targets/CMSIS-OS/ChibiOS/MBN_QUAIL/common/usbcfg.c [hunk 1]
```diff
@@ -87,7 +88,7 @@ static const uint8_t vcom_device_descriptor_data[18] = {
                          0x0200,        /* bcdDevice.                       */
                          1,             /* iManufacturer.                   */
                          2,             /* iProduct.                        */
-                         3,             /* iSerialNumber.                   */
+                         0,             /* iSerialNumber.                   */
```
Comment (josesimoes) [inline targets/CMSIS-OS/ChibiOS/MBN_QUAIL/common/usbcfg.c]: With this you are telling that the device serial number is at index = 0, which is not correct. It's on 3. 
At index 0 is the language description.
These are stored at `vcom_strings[]`.
This is the reason why Quail is not showing on nf-ANT. The condition for a COM port device to be recognized as a "nanoFramework device" is that it's serial number starts with `NANO_` (see `#define USB_STRING_SERIAL_NUMBER...`)

File: targets/CMSIS-OS/ChibiOS/MBN_QUAIL/common/usbcfg - Copie.c [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+
+/*
+ * Serial over USB driver configuration.
+ */
+const SerialUSBConfig serusbcfg = {
+  &USBD1,
+  USBD1_DATA_REQUEST_EP,
+  USBD1_DATA_AVAILABLE_EP,
+  USBD1_INTERRUPT_REQUEST_EP
+};
```
Comment (josesimoes) [inline targets/CMSIS-OS/ChibiOS/MBN_QUAIL/common/usbcfg - Copie.c]: Please make sure to always have an empty line at the very end of each file to keep GitHub diff tool happy. 😉

File: targets/CMSIS-OS/ChibiOS/MBN_QUAIL/common/usbcfg.c (hunk 1 above)
Comment (MikroBusNet) [inline targets/CMSIS-OS/ChibiOS/MBN_QUAIL/common/usbcfg.c]: According to [this site](http://www.beyondlogic.org/usbnutshell/usb5.shtml) :
"Three string descriptors exist to provide details of the manufacturer, product and serial number. There is no requirement to have string descriptors. If no string descriptor is present, a index of zero should be used."

Hence the 0 to not show the serial number. I thought I did not need a serial number. It seems I am wrong regarding nf way of doing things. I will fix this.

File: targets/CMSIS-OS/ChibiOS/MBN_QUAIL/common/usbcfg - Copie.c (hunk 2 above)
Comment (MikroBusNet) [inline targets/CMSIS-OS/ChibiOS/MBN_QUAIL/common/usbcfg - Copie.c]: Understood !

**Comment by MikroBusNet** [comment]
It seems I have again messed up things between master upstream and my local repo/branches :(
I will once again erase everything and start from scratch with this very nice and very user-friendly Github thing...

**Comment by josesimoes** [comment]
No need to start over! Just "git pull upstream master " on your master branch and then "git pull master" on your "quail " branch. After resolving the likely conflicts your should be good to go. 😄

**Comment by MikroBusNet** [comment]
Sorry to be as dumb as I am but how would you resolve that : 

screen shot 003

**Comment by MikroBusNet** [comment]
Please delete this PR. I am cooking a new one that will hopefully follow its way to the repo, this time :)

**Comment by josesimoes** [comment]
Will do.

Hopefully you'll have the saying on your side "third time's the charm" 😉

**Comment by josesimoes** [comment]
BTW, you may want to hold your next PR a little bit until #149 is merged. 

It will make you life easier because:
- you won't have to change FindChibiOS.cmake anymore
- you don't have to add the QUAIL board support in the overlay
- you just have to add those board.c and board.h right into the target board folder.

**Pull request closed without merging**